library/psa_crypto.c

```diff
@@ -5076,15 +5076,41 @@ static psa_status_t psa_tls12_prf_psk_to_ms_input(
 }
 #endif /* MBEDTLS_MD_C */
 
+static int psa_key_derivation_check_input_type(
+    psa_key_derivation_step_t step,
+    psa_key_type_t key_type )
+{
+    switch( step )
+    {
+        case PSA_KEY_DERIVATION_INPUT_SECRET:
+            if( key_type == PSA_KEY_TYPE_DERIVE || key_type == 0 )
```

Comment:
What does a key type of zero mean? Why not use a define for anything it is?

Reply:
0 is not a valid key type. Here it's used to mean that the input is a direct string rather than a key object.

Reply:
A separate define signalling that (to be used in test data, since it's ambiguous there too) would be useful. If not - at least a comment here and in tests.

Reply:
Done.